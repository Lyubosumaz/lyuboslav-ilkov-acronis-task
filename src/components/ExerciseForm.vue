<template>
    <section>
        <h3>Add new exercise:</h3>

        <el-form
            ref="form"
            :model="form"
            label-width="120px"
            class="exersice-form-wrapper"
        >
            <el-form-item label="Exercise name">
                <el-input v-model="form.name"></el-input>
            </el-form-item>

            <el-form-item label="Exercise sets">
                <el-input v-model="form.sets"></el-input>
            </el-form-item>

            <el-button @click="addExercise(form.name, form.sets)"
                >Add Exercise</el-button
            >
        </el-form>
    </section>
</template>

<script>
import { mapGetters, mapActions } from 'vuex';
import { mockFetchREST } from '../utils/mockFetch';

export default {
    name: 'ExersiceForm',
    props: {
        workoutId: Number,
    },
    data() {
        return {
            form: {
                name: '',
                sets: '',
            },
        };
    },
    methods: {
        ...mapActions(['createExercise']),
        addExercise(exerciseName, exerciseSets) {
            this.createExercise([
                this.workoutId,
                {
                    name: exerciseName,
                    sets: exerciseSets,
                },
            ]);

            this.form.name = '';
            this.form.sets = '';

            // mockFetch 'POST'
            mockFetchREST(this.getWorkouts);
        },
    },
    computed: mapGetters(['getWorkouts']),
};
</script>

<style scoped>
.exersice-form-wrapper {
    display: flex;
    justify-content: space-between;
}
.exersice-form-wrapper > div {
    margin: 0;
}
</style>
