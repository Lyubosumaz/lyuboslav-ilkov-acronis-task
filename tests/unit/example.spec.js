import { mount, createLocalVue } from '@vue/test-utils';
import ElementUI from 'element-ui';
import ExerciseForm from '@/components/ExerciseForm.vue';

describe('raw ExerciseForm', () => {
    it('has data', () => {
        expect(typeof ExerciseForm.data).toBe('function');
    });
});

describe('mounted ExerciseForm', () => {
    const localVue = createLocalVue();
    localVue.use(ElementUI);
    const wrapper = mount(ExerciseForm, { localVue });

    test('is a Vue with ElementUI instance', () => {
        expect(wrapper.vm).toBeTruthy();
    });

    it('renders the correct markup h3', () => {
        expect(wrapper.html()).toContain('<h3>Add new exercise:</h3>');
    });
});

describe('snapshot ExerciseForm', () => {
    const localVue = createLocalVue();
    localVue.use(ElementUI);
    const wrapper = mount(ExerciseForm, { localVue });

    it('renders correctly to match snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
});
