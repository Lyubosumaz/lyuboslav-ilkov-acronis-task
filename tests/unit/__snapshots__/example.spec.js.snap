// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot ExerciseForm renders correctly to match snapshot 1`] = `
<section>
  <h3>
    Add new exercise:
  </h3>
   
  <form
    class="el-form exersice-form-wrapper"
  >
    <div
      class="el-form-item"
    >
      <label
        class="el-form-item__label"
        style="width: 120px;"
      >
        Exercise name
      </label>
      <div
        class="el-form-item__content"
        style="margin-left: 120px;"
      >
        <div
          class="el-input"
        >
          <!---->
          <input
            autocomplete="off"
            class="el-input__inner"
            type="text"
          />
          <!---->
          <!---->
          <!---->
          <!---->
        </div>
        <transition-stub
          name="el-zoom-in-top"
        >
          <!---->
        </transition-stub>
      </div>
    </div>
     
    <div
      class="el-form-item"
    >
      <label
        class="el-form-item__label"
        style="width: 120px;"
      >
        Exercise sets
      </label>
      <div
        class="el-form-item__content"
        style="margin-left: 120px;"
      >
        <div
          class="el-input"
        >
          <!---->
          <input
            autocomplete="off"
            class="el-input__inner"
            type="text"
          />
          <!---->
          <!---->
          <!---->
          <!---->
        </div>
        <transition-stub
          name="el-zoom-in-top"
        >
          <!---->
        </transition-stub>
      </div>
    </div>
     
    <button
      class="el-button el-button--default"
      type="button"
    >
      <!---->
      <!---->
      <span>
        Add Exercise
      </span>
    </button>
  </form>
</section>
`;
